src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferenceCache.cs

```diff
@@ -19,36 +19,24 @@
 
 namespace Microsoft.CodeAnalysis.FindSymbols
 {
-    // TODO : this can be all moved down to compiler side.
     internal static class FindReferenceCache
     {
-        private static readonly ReaderWriterLockSlim s_gate = new();
-        private static readonly Dictionary<SemanticModel, Entry> s_cache = new();
+        private static readonly ConditionalWeakTable<SemanticModel, Entry> s_cache = new();
```

Comment:
⚠️ Gen and I were observing cases where the target in CWT was not quickly reclaimed by the GC. We should avoid putting large object graphs here since it will force them to Gen 2 GC. In this case `Entry` contains references to both green trees and symbol trees.

Reply:
going to gen-2 may be desirable here.  For example, a core FAR use case is doing many find-refs in a row.  It's a goal to cache much/all of hte intermediate computation so that subsequent FAR operations (without intervening edits) is fast.  We do this caching heavily at many levels of FAR and associated components (like 'dependent type walking').

Reply:
CodeLens hits this path pretty hard. It could lead to unexpectedly high working set sizes during longer editing sequences.

Reply:
or, we don't cache, and then codelens chews through more CPU as this information is computed/dropped/computed/dropped :)

it's a design goal of FAR to allow N hits of it (including serially) that can benefit from prior computation so that intermediary results can be used.  That's a way that services use it (including users, or features like CodeLens). 

If working set is an issue, and we're seeing CWTs exacerbate that, then i think we have a general problem as this is a common thing we do to cache expensive work.

Reply:
My only real concern here is the SymbolInfo cache. The other cached items are smaller items similar to what we see in other caches.

Reply:
Do we know why the GC isn't good at reclaiming things here?  Note that this code should only run in OOP, where i would expect .net core to be good about these.  @genlu can you shed any light on this?